org.eclipse.xtext.xbase.ui/src/org/eclipse/xtext/xbase/ui/refactoring/XbaseReferenceUpdater.java

```diff
@@ -109,7 +109,7 @@ protected void createReferenceUpdate(EObject referringElement, URI referringReso
 				JvmDeclaredType type = (JvmDeclaredType) newTargetElement;
 
 				ImportAwareUpdateAcceptor importAwareUpdateAcceptor = (ImportAwareUpdateAcceptor) updateAcceptor;
-				importAwareUpdateAcceptor.removeImport(type, isStatic, isExtension, memberName);
+				importAwareUpdateAcceptor.removeImport(importedType, isStatic, isExtension, memberName);
 				importAwareUpdateAcceptor.acceptImport(type, isStatic, isExtension, memberName);
```

Comment:
In the light of this change I find the line 137 interesting where we add the importedType again after having it removed in line 127. Something seems to be off in the reference updater.
It would be great if we had a test for this probably submitted against xtend-xtend

Reply:
FTR: When I tried this with a rename op in Xtend, the importedType and the given type are exactly the same object. So the change would not make a difference.

Reply:
@szarnekow when i tried with xtend i could not reproduce.